app/assets/javascripts/map.js

```diff
@@ -0,0 +1,210 @@
+//= require polyline
+var map;
+
+var directionsService = new google.maps.DirectionsService();
+
+function initialize() {
+  directionsDisplay = new google.maps.DirectionsRenderer();
+
+  var mapOptions = {
+    zoom: 4,
+    mapTypeId: google.maps.MapTypeId.ROADMAP
+  };
+
+  map = new google.maps.Map(document.getElementById('map_canvas'), mapOptions);
+  directionsDisplay.setMap(map);
+  route = new Route();
+  route.calculateRoute();
+
+}
+
+//-------Route----------
+
+var Route = function(){
+  this.start = new google.maps.LatLng(41.953819, -87.654750), // Chicago for now
+  this.end = new google.maps.LatLng(38.637548, -90.205010), // St. Louis for now
+  this.waypts = this.getWaypoints();
+};
+
+Route.prototype.getWaypoints = function(){
+  var waypts = [];
+  var checkboxArray = document.getElementById('waypoints');
```

Comment:
You might consider having a `RouteView` object that is responsible for accessing the DOM and translating that into data and a `Route` object that's responsible for calculating the route and holding data.